### PR TITLE
fix: improve error handling for invalid URL submission

### PR DESCRIPTION
**Pull request by devamitranjan**
# Motivation

Added the URL regex to ensure the validation before submitting the data.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (Storybook)
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by devamitranjan** [comment]
[Screencast from 24-09-24 02:13:58 AM IST.webm](https://github.com/user-attachments/assets/3718dae9-0fe1-4172-a9af-670fc10d8747)

**Comment by Emnaghz** [comment]
Thank you @devamitranjan for your contribution :tada: :partying_face: :rocket:

File: frontend/src/components/Menu/MenuDialog.tsx [hunk 1]
```diff
@@ -66,7 +67,8 @@ export const MenuDialog: FC<MenuDialogProps> = ({
     url: {
       required: t("message.url_is_invalid"),
       validate: (value: string = "") =>
-        isAbsoluteUrl(value) || t("message.url_is_invalid"),
+        (isAbsoluteUrl(value) && URL_REGEX.test(value)) ||
```
Comment (marrouchi) [inline frontend/src/components/Menu/MenuDialog.tsx:70]: Thank you @devamitranjan 🙏 

Although it looks good, I suggest we use rather the URL class : 

```
function isAbsoluteUrl(value: string) {
  let url;
  
  try {
    url = new URL(value);
  } catch (_) {
    return false;  
  }

  return url.protocol === "http:" || url.protocol === "https:";
}
```

Let me know what you think.

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: frontend/src/components/Menu/MenuDialog.tsx (hunk 1 above)
Comment (devamitranjan) [reply frontend/src/components/Menu/MenuDialog.tsx]: @marrouchi Thanks for the suggestion. Did the changes.

**Pull request merged**